medcat/utils/meta_cat/data_utils.py

```diff
@@ -180,9 +186,38 @@ def encode_category_values(data: Dict, existing_category_value2id: Optional[Dict
         category_value2id = {}
 
     category_values = set([x[2] for x in data])
-    for c in category_values:
-        if c not in category_value2id:
-            category_value2id[c] = len(category_value2id)
+
+    # If categoryvalue2id is pre-defined, then making sure it is same as the labels found in the data
+    if len(category_value2id) != 0:
+        if set(category_value2id.keys()) != category_values:
+            # if categoryvalue2id doesn't match the labels in the data, then 'class_name_map' has to be defined to check for variations
+            if len(class_name_map) != 0:
+                updated_category_value2id = {}
+                for _class in category_value2id.keys():
+                    if _class in category_values:
+                        updated_category_value2id[_class] = category_value2id[_class]
+                    else:
+                        found_in = [sub_map for sub_map in class_name_map if _class in sub_map][0]
+                        if len(found_in) != 0:
+                            class_name_matched = [label for label in found_in if label in category_values][0]
```

Comment:
If there's no existing category names defined in the "map", this will fail with an `IndexError`.
If there's more than one existing category name in this "map", I think and exception should be raised as it'd be a misconfiguration.